pkg/client/kubelet.go

```diff
@@ -135,7 +136,43 @@ func (c *HTTPKubeletClient) GetPodStatus(host, podNamespace, podID string) (api.
 }
 
 func (c *HTTPKubeletClient) HealthCheck(host string) (probe.Result, error) {
-	return httprobe.DoHTTPProbe(fmt.Sprintf("%s/healthz", c.url(host)), c.Client)
```

Comment:
The hostname used by the apiserver should be available in the `Host` header of the request received by kubelet. pkg/healthz isn't really developed enough to do smart healthchecks like this yet though, and not sure if there are plans to expand it's ability. I think the check on the client side makes sense here.